src/deluge/model/instrument/non_audio_instrument.cpp

```diff
@@ -34,6 +35,11 @@ void NonAudioInstrument::renderOutput(ModelStack* modelStack, StereoSample* star
 
 		if (activeInstrumentClip->arpSettings.mode != ArpMode::OFF) {
 			uint32_t gateThreshold = activeInstrumentClip->arpeggiatorGate + 2147483648;
+			uint32_t ratchetProbability = activeInstrumentClip->arpeggiatorRatchetProbability + 2147483648;
+			uint32_t ratchetAmount = activeInstrumentClip->arpeggiatorRatchetAmount + 2147483648;
+			uint32_t sequenceLength =
+			    (((int64_t)activeInstrumentClip->arpeggiatorSequenceLength + 2147483648) * kMaxMenuValue + 2147483648)
+			    >> 32;
```

Comment:
```suggestion
			uint32_t gateThreshold = activeInstrumentClip->arpeggiatorGate + 2147483648U;
			uint32_t ratchetProbability = activeInstrumentClip->arpeggiatorRatchetProbability + 2147483648U;
			uint32_t ratchetAmount = activeInstrumentClip->arpeggiatorRatchetAmount + 2147483648U;
			uint32_t sequenceLength =
			    (((int64_t)activeInstrumentClip->arpeggiatorSequenceLength + 2147483648) * kMaxMenuValue + 2147483648U)
			    >> 32;
```

I think these calculations can overflow otherwise? It's unclear how `gateThreshold` was working before.

Reply:
The gateThreshold line has been like that from the beginnings so I just did the same with my new parameters. I haven't experienced any buggy behavior even setting parameter values from minimum to maximum values